apps/governance/src/components/web3-connector/web3-connector.tsx

```diff
@@ -52,23 +49,61 @@ export function Web3Connector({
 interface Web3ContentProps {
   children: ReactElement;
   appChainId: number;
-  setDialogOpen: (isOpen: boolean) => void;
 }
 
 export const Web3Content = ({ children, appChainId }: Web3ContentProps) => {
+  const { appState, appDispatch } = useAppState();
   const { connector, chainId } = useWeb3React();
+  const [previousChainId, setPreviousChainId] = useState(chainId);
   const error = useWeb3ConnectStore((store) => store.error);
   const disconnect = useWeb3Disconnect(connector);
 
+  const showDisconnectNotice = useCallback(
+    (isVisible: boolean) =>
+      appDispatch({
+        type: AppStateActionType.SET_DISCONNECT_NOTICE,
+        isVisible,
+      }),
+    [appDispatch]
+  );
+
   useEffect(() => {
     if (connector?.connectEagerly) {
       connector.connectEagerly();
     }
-    // wallet connect doesnt handle connectEagerly being called when connector is also in the
+    // wallet connect doesn't handle connectEagerly being called when connector is also in the
     // deps array.
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
+  useEffect(() => {
+    if (chainId !== undefined) {
+      // We use this to detect when the user switches networks.
+      setPreviousChainId(chainId);
+
+      if (chainId !== appChainId) {
+        disconnect();
+
+        // If the user was previously connected, show the disconnect explanation notice.
+        if (previousChainId !== undefined && !appState.disconnectNotice) {
+          showDisconnectNotice(true);
+        }
+      } else {
+        if (appState.disconnectNotice) {
+          showDisconnectNotice(false);
+        }
+      }
+    }
```

Comment:
nitpick (non-blocking): would probably makes this an else if rather than an else and an if